From 5c9fbdd64313607e0aacf4a5248a1873a4514cdd Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Mon, 19 Aug 2024 19:40:39 +0200
Subject: [PATCH 2/5] wifi: mt7915: fix wcid allocation leak

Signed-off-by: Felix Fietkau <nbd@nbd.name>
---
 mt7915/main.c | 1 +
 1 file changed, 1 insertion(+)

--- a/mt7915/main.c
+++ b/mt7915/main.c
@@ -292,6 +292,7 @@ static void mt7915_remove_interface(stru
 
 	mt7915_mcu_add_bss_info(phy, vif, false);
 	mt7915_mcu_add_sta(dev, vif, NULL, false);
+	mt76_wcid_mask_clear(dev->mt76.wcid_mask, mvif->sta.wcid.idx);
 
 	mutex_lock(&dev->mt76.mutex);
 	mt76_testmode_reset(phy->mt76, true);
