From caa62d58ab916fe5358e751adda9393be4cf0398 Mon Sep 17 00:00:00 2001
From: tarori <ritarot634@gmail.com>
Date: Tue, 23 Jan 2024 04:27:34 +0900
Subject: [PATCH] Disable prefetch for arm64

---
 arch/arm64/include/asm/processor.h | 2 --
 1 file changed, 2 deletions(-)

--- a/arch/arm64/include/asm/processor.h
+++ b/arch/arm64/include/asm/processor.h
@@ -278,13 +278,11 @@ extern struct task_struct *cpu_switch_to
 #define ARCH_HAS_PREFETCH
 static inline void prefetch(const void *ptr)
 {
-	asm volatile("prfm pldl1keep, %a0\n" : : "p" (ptr));
 }
 
 #define ARCH_HAS_PREFETCHW
 static inline void prefetchw(const void *ptr)
 {
-	asm volatile("prfm pstl1keep, %a0\n" : : "p" (ptr));
 }
 
 #define ARCH_HAS_SPINLOCK_PREFETCH
